rs/bitcoin/kyt/src/main.rs

```diff
@@ -94,6 +95,23 @@ fn post_upgrade(arg: KytArg) {
     }
 }
 
+#[ic_cdk::query(hidden = true)]
+fn http_request(req: http::HttpRequest) -> http::HttpResponse {
+    if req.path() == "/metrics" {
+        // TODO: metrics
+        unimplemented!()
+    } else if req.path() == "/dashboard" {
+        use askama::Template;
+        let dashboard = dashboard::dashboard().render().unwrap();
+        http::HttpResponseBuilder::ok()
+            .header("Content-Type", "text/html; charset=utf-8")
+            .with_body_and_content_length(dashboard)
+            .build()
```

Comment:
The number of entries in the cache ensures that the rendered dashboard can never exceed 2MiB, correct? Would it make sense to have a proptest for that or is that overkill?

Reply:
Actually I have not thought of this. Not a bad idea to add such a test, but it can only be an estimate, since some transactions have more inputs than others.

Reply:
Added a test, and it shows the rendered HTML exceeds 2MB! Had to reduce `MAX_FETCH_TX_ENTRIES` to 3000 in order for this check to succeed. I guess this is an arbitrary setting anyway, and 3000 still sounds plenty...

Reply:
ha that's a nice finding! Thanks a lot for the test and adjusting the parameter!

Reply:
The bound is surprisingly low. Why 3000?
Roughly 90% of transactions are below 400 bytes. If we take that as a conservative average size, we get `2^21 / 400 = 5242.88`, i.e., 5000 addresses would fit, right?